### PR TITLE
Make it build with Melange

### PR DESCRIPTION
**Pull request by anmonteiro**
This sets up a build with [Melange](https://github.com/melange-re/melange/).

- af208e5 just sets up dependencies (upgrades bisect_ppx to get ppxlib support, upgrades bastet to 2.0)
- b7fcc55 makes the project build with Melange:
  1. disables warning 60 (unused module) – there were some of these in functor arguments; OCaml 4.12 started to emit some more of these warnings
  2. removes `\` (backslash) escapes in the `bsconfig.json` arguments to bisect_ppx. Dune treats `\` as a first class citizen, so these escapes aren't needed.
  3. Renames `<#>` (`flipMap`) to `<$$>`: OCaml 4.12 apparently disallows any infix operator that has `#` in its name. I chose `<$$>` randomly, and I'm not aware of any conventions for `flipMap`. Suggestions appreciated for a better name here.

### CODE REVIEW

**Comment by mlms13** [comment]
Thanks so much for doing this! I was just reaching out to @johnhaley81 to see if there was anything I could do to help this effort, and it looks like it's done! I should be able to review sometime over the next few days (definitely feel free to ping me on Discord if I don't by next weekend).

For the `flipMap` infix, it looks like Bastet uses `<@>`. I don't have a strong preference, but maybe we do the same for consistency.

**Comment by anmonteiro** [comment]
Thanks @mlms13!

I thought about `<@>` too, but sounds like it's taken by [`flap`](https://github.com/reazen/relude/blob/master/src/extensions/Relude_Extensions_Functor.re#L72)

**Comment by anmonteiro** [comment]
Do you have thoughts on CI? Should I try to set up a build with both ReScript and Melange, or just one of them?

**Review by andywhite37 — APPROVED** [review]
This all looks good to me! Thanks for doing the work to get this upgraded - long live melange! As far as CI build - if you wanted to submit a build for `melange` that would be appreciated - I haven't really kept up with how that would be done in these different ecosystems. We can also check out how it's done in other projects if you know of any we could copy.

I'm fine with `<$$>` - I think `<#>` was somewhat arbitrary too. I'll let @mlms13 take a look too before we merge.

**Comment by mlms13** [comment]
> I thought about <@> too, but sounds like it's taken by flap

Ah yeah, good call. I think we maybe picked `<#>` because that's what Purescript does. Purescript uses `<@>` for `flap`, which is probably where we got that one too, so I guess we should leave that one unchanged. Since `<#>` is no longer an option, I don't really have much of a preference; `<$$>` is good with me.

> Do you have thoughts on CI?

I think CI was broken for other reasons before this PR. Specifically, I think the OCaml base image we were using for the Giuthub Actions went away, and we couldn't switch to the better-supported OCaml image because it didn't support 4.06. But now we should be able to use a good OCaml version, so that's awesome! 🤗 No need to make that change as part of this PR, though... CI fixes can be tackled separately.

File: __tests__/extensions/Relude_Extensions_Ord_test.re
Comment (mlms13) [inline __tests__/extensions/Relude_Extensions_Ord_test.re:3]: I wonder if this should be `open! Relude.Globals` instead of `open !Relude.Globals`, but if it works this way, I'm equal parts confused an indifferent. 😁

File: __tests__/js/Relude_Js_Json_test.re
Comment (mlms13) [inline __tests__/js/Relude_Js_Json_test.re:795]: Probably want to change the `<#>` in the comment here as well.

**Review by mlms13 — COMMENTED** [review]
A couple super minor comments, but overall, this looks really great and basically ready to merge.

Since this opens us up to supporting newer OCaml features that will be incompatible with the Rescript compiler (specifically looking at you, letops), we probably want to have a conversation about how we want to manage versions from here on out. Maybe we make a 1.0 release that is the final Relude release to officially work with the Rescript compiler, then  we turn our attention to a 2.0 branch that leverages some of the newer OCaml goodies.

But it looks like this PR probably still works with the Rescript compiler, so we might not need to make that call right now.

File: __tests__/extensions/Relude_Extensions_Ord_test.re
Comment (anmonteiro) [reply __tests__/extensions/Relude_Extensions_Ord_test.re:3]: This is probably copy pasta. will fix

File: __tests__/js/Relude_Js_Json_test.re
Comment (anmonteiro) [reply __tests__/js/Relude_Js_Json_test.re:795]: great point. I thought I'd gotten all of them!

**Comment by anmonteiro** [comment]
Just fixed the code review items, thanks for the review all!

**Review by mlms13 — APPROVED** [review]

**Comment by mlms13** [comment]
Oops, we never merged this! Let's party, yeah?!

**Pull request merged**
